mcsema/BC/Function.cpp

```diff
@@ -260,6 +289,132 @@ static llvm::BasicBlock *GetOrCreateBlock(TranslationContext &ctx,
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+      auto lpad = ir.CreateLandingPad(exn_type, 1, "cleanup.lpad");
+#else
+      auto personality = gModule->getFunction("__gxx_personality_v0");
+      auto lpad = ir.CreateLandingPad(exn_type, personality, 1, "cleanup.lpad");
+#endif
+      lpad->setCleanup(true);
+
+      auto exctn_0  = ir.CreateExtractValue(lpad, 0);
+      auto store_loc_0 = ir.CreateAlloca(exctn_0->getType());
+      ir.CreateStore(exctn_0, store_loc_0);
+
+      auto exctn_1 = ir.CreateExtractValue(lpad, 1);
+      auto store_loc_1 = ir.CreateAlloca(exctn_1->getType());
+      ir.CreateStore(exctn_1, store_loc_1);
+
+      auto lp_entry = ctx.ea_to_block[entry->lp_ea];
+     // landing_bb->dump();
+      if (nullptr != lp_entry) {
+        ir.CreateBr(lp_entry);
+        ctx.lp_to_block[entry->lp_ea] = landing_bb;
+      }
+    } else if (entry->action) {
+      std::stringstream ss;
+      ss << "catch_landingpad_" << std::hex << entry->lp_ea;
+      llvm::BasicBlock *landing_bb = llvm::BasicBlock::Create(
+          *gContext, ss.str(), ctx.lifted_func);
+      llvm::IRBuilder<> ir(landing_bb);
+      auto num_clauses = entry->ttype.size();
+      auto exn_type = llvm::StructType::get(llvm::Type::getInt8PtrTy(*gContext),
+                                            llvm::Type::getInt32Ty(*gContext), nullptr);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+      auto lpad = ir.CreateLandingPad(exn_type, static_cast<unsigned int>(num_clauses), "catch.lpad");
+#else
+      auto personality = gModule->getFunction("__gxx_personality_v0");
+      auto lpad = ir.CreateLandingPad(exn_type, personality, static_cast<unsigned int>(num_clauses), "catch.lpad");
+#endif
+      unsigned int catch_all = false;
+
+      //catch (...)
+      // Set tne clause to catch all kind of exceptions
+      /**
+       * The type indices in exception table are reversed for some of the binaries. In such case, the
+       * wrong exception handler may get called.
+       * TODO: Fix the wrong indices in the exception table.
+       */
+      for (auto type = entry->ttype.rbegin(); type != entry->ttype.rend(); type++) {
+        if ((*type)->ea) {
+          lpad->addClause(gModule->getGlobalVariable((*type)->name));
+        } else {
+          catch_all = true;
+        }
+      }
+
+      if (catch_all) {
+        lpad->addClause(llvm::Constant::getNullValue(ir.getInt8PtrTy()));
+      }
+
+      std::vector<llvm::Value *> args(2);
+      llvm::Type* args_type[] = {llvm::Type::getInt64Ty(*gContext), llvm::Type::getInt64Ty(*gContext)};
+
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+      args[0] = ir.CreateLoad(llvm::Type::getInt64Ty(*gContext), cfg_func->rsp_var);
+      args[1] = ir.CreateLoad(llvm::Type::getInt64Ty(*gContext), cfg_func->rbp_var);
+#else
+      args[0] = ir.CreateLoad(cfg_func->rsp_var, true);
+      args[1] = ir.CreateLoad(cfg_func->rbp_var, true);
+#endif
+
+      auto handler = gModule->getFunction("__mcsema_exception_ret");
```

Comment:
Make a `static` function that returns the `llvm::Function *` for `__mcsema_exception_ret`, and call it here. That function should have a `static llvm::Function *exception_ret = nullptr;` var in it.